slack_sdk/models/blocks/block_elements.py

```diff
@@ -1381,6 +1405,186 @@ def __init__(
+
+
+# -------------------------------------------------
+# Number Input Element
+# -------------------------------------------------
+
+
+class NumberInputElement(InputInteractiveElement):
+    type = "number_input"
+
+    @property
+    def attributes(self) -> Set[str]:
+        return super().attributes.union(
+            {
+                "initial_value",
+                "is_decimal_allowed",
+                "min_value",
+                "max_value",
+                "dispatch_action_config",
+            }
+        )
+
+    def __init__(
+        self,
+        *,
+        action_id: Optional[str] = None,
+        is_decimal_allowed: Optional[bool] = None,
+        initial_value: Optional[str] = None,
+        min_value: Optional[str] = None,
+        max_value: Optional[str] = None,
+        dispatch_action_config: Optional[Union[dict, DispatchActionConfig]] = None,
+        focus_on_load: Optional[bool] = None,
+        placeholder: Optional[Union[str, dict, TextObject]] = None,
+        **others: dict,
+    ):
+        """
+        https://api.slack.com/reference/block-kit/block-elements#number
+
+        Args:
+            action_id (required): An identifier for the input value when the parent modal is submitted.
+                You can use this when you receive a view_submission payload to identify the value of the input element.
+                Should be unique among all other action_ids in the containing block.
+                Maximum length for this field is 255 characters.
+            is_decimal_allowed (required): Decimal numbers are allowed if is_decimal_allowed= true, set the value to
+                false otherwise.
+            initial_value: The initial value in the number input when it is loaded.
+            min_value: The minimum value, cannot be greater than max_value.
+            max_value: The maximum value, cannot be less than min_value.
+            dispatch_action_config: A dispatch configuration object that determines when
+                during text input the element returns a block_actions payload.
+            focus_on_load: Indicates whether the element will be set to auto focus within the view object.
+                Only one element can be set to true. Defaults to false.
+            placeholder: A plain_text only text object that defines the placeholder text shown
+                in the plain-text input. Maximum length for the text in this field is 150 characters.
+        """
+        super().__init__(
+            type=self.type,
+            action_id=action_id,
+            placeholder=TextObject.parse(placeholder, PlainTextObject.type),
+            focus_on_load=focus_on_load,
+        )
+        show_unknown_key_warning(self, others)
+
+        self.initial_value = str(initial_value) if initial_value else None
+        self.is_decimal_allowed = is_decimal_allowed
+        self.min_value = str(min_value) if min_value else None
+        self.max_value = str(max_value) if max_value else None
```

Comment:
0 is falsy in Python. These parts should be more explicit:

```python
>>> if 0:
...     print("true")
... else:
...     print("false")
...
false
```

```suggestion
        self.min_value = str(min_value) if min_value is not None else None
        self.max_value = str(max_value) if max_value is not None else None
```

Reply:
Also, if we accept non-string values here, the type hints for arguments need to be updated too

Reply:
Ahh yess god catch 🥇 did not think about the 0 falsy

Reply:
I've added type `Optional[Union[int, float, str]]` for initial_value, min_value, max_value